src/ArmTemplates/Common/API/Clients/Abstractions/ApiClientBase.cs

```diff
@@ -24,37 +25,55 @@ public abstract class ApiClientBase
 
         protected AzureCliAuthenticator Auth { get; private set; } = new AzureCliAuthenticator();
 
+        protected Dictionary<string, HttpMethod> httpMethodsMap;
+
+        public const string HTTP_GET_METHOD = "get";
+        public const string HTTP_POST_METHOD = "post";
+
         public ApiClientBase(string baseUrl = null) 
         {
             if (!string.IsNullOrEmpty(baseUrl))
             {
                 this.BaseUrl = baseUrl;
             }
+
+            this.httpMethodsMap = new Dictionary<string, HttpMethod>();
+            this.httpMethodsMap.Add(HTTP_GET_METHOD, HttpMethod.Get);
+            this.httpMethodsMap.Add(HTTP_POST_METHOD, HttpMethod.Post);
         }
 
-        protected async Task<string> CallApiManagementAsync(string azToken, string requestUrl)
+        protected async Task<string> CallApiManagementAsync(string azToken, string requestUrl, bool useCache = true, string method = HTTP_GET_METHOD)
```

Comment:
Answering to the question regarding HTTPMethod Enum, in order to not break any previous functionality, I needed to have default value GET, since default parameter value should be available at compile time, But HttpMethod is not, that is why it was introduced. 
Regarding the suggestion to split Post Method and Get methods separately:

1. In such way, we will have two pretty identical methods to each other with difference in methods and additional parameter (body)
2. In both solution we are setting the limit on usage of other http methods. 
3. If be honest I do not see the problem from adding optional parameter for the method, in case it is needed. In both solution in case there will be a need to pass the headers (for example) we will add additional parameters to the method but the difference is that we will be doing it for one method or two.
Open to discussion :)